components/raftstore-v2/src/raft/apply.rs

```diff
@@ -106,6 +139,16 @@ impl<EK: KvEngine, R> Apply<EK, R> {
         &mut self.region_state
     }
 
+    #[inline]
+    pub fn set_region_state(&mut self, region_state: RegionLocalState) {
+        self.region_state = region_state;
+    }
+
+    #[inline]
+    pub fn tablet(&mut self) -> Option<EK> {
```

Comment:
The dynamics between `self.tablet` and `remote_tablet` is confusing. From the code I have no idea how to choose between `self.tablet`, `remote_tablet.cache()`, `remote_tablet.latest()`.

Reply:
I don't know why we need filed `tablet`. The code before uses it to generate write batch if self.write_batch is none in apply thread. But raftstore can update the remote_tablet (ex: when parent peer change from split_x_y to _x_y) so in apply thread, it should always check whether the tablet has a newer version. @BusyJay

Reply:
No, `remote_tablet` is only used for publishing new tablets. Check `publish_tablet`. You should always use `self.tablet` instead of querying remote.